cmd/virtual-kubelet/commands/root/root.go

```diff
@@ -160,12 +161,23 @@ func runRootCommand(ctx context.Context, c Opts) error {
 	}
 	defer cancelHTTP()
 
+	started := make(chan struct{})
 	go func() {
-		if err := vk.Run(ctx); err != nil && errors.Cause(err) != context.Canceled {
+		if err := vk.Run(ctx, started); err != nil && errors.Cause(err) != context.Canceled {
 			log.G(ctx).Fatal(err)
 		}
 	}()
 
+	// TODO: Consider waiting for up to startup timeout seconds for this to close
+	//
+	// currently, startup timeout is just used in waiting for the VK to start. It has its own timeout
+	// mechanism, so this is redundant if we add timeouts here.
+	//
+	// Even though it wont close the channel if the VK fails to start, it will cause a fatal log
+	// message to be emitted, shutting down the process
+	if c.WaitForStartup {
+		<-started
```

Comment:
I needed something similar for #586 
I decided to add a `.Ready()` function instead of accepting a channel.
WDYT? https://github.com/virtual-kubelet/virtual-kubelet/pull/586/files#diff-c334be5faaed7b87440b6090f6dd3f6fR96

Reply:
I like the Ready() approach better, and I'm okay with changing it. I just feel weird that it introduces state to the struct that's not necessary, and IMHO, unless there is a good reason to add members to a struct, don't do it.